### PR TITLE
Support stream level configuration on stream creation

### PR DESCRIPTION
**Pull request by LaPetiteSouris**
This PR is to add custom stream level configuration in the API, so that upon a CreateStream request, the client has the possibility to customize the stream configuration.

Required [PR](https://github.com/liftbridge-io/liftbridge-api/pull/26)

Related [issue](https://github.com/liftbridge-io/liftbridge/issues/75)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +90,19 @@ in the same group, messages will be balanced among them.
 // all the brokers in the cluster. ErrStreamExists is returned if a stream with
 // the given name already exists.
 client.CreateStream(context.Background(), "foo.*", "my-stream", lift.MaxReplication())
+
+// Create a stream attached to the NATS subject "foo.*" that is replicated to
+// all the brokers in the cluster. ErrStreamExists is returned if a stream with
+// the given name already exists.
+// This also set the retention.max.age for the created stream
+client.CreateStream(context.Background(), subject, name, lift.RetentionMaxAge(time.Nanosecond))
```
Comment (tylertreat) [inline README.md:98]: Probably should set `RetentionMaxAge` to a more realistic value, such as 1 hour. I've found people often copy/paste examples and using a nanosecond for retention wouldn't be a great idea. :)

File: client.go [hunk 2]
```diff
@@ -88,6 +95,30 @@ type StreamOptions struct {
...
 	Partitions int32
+
+	// The maximum size a stream's log can grow to, in bytes, before we will discard
+	// old log segments to free up space. A value of 0 indicates no limit.
+	RetentionMaxBytes int64
+	// The maximum size a stream's log can grow to, in number of messages, before we will discard
+	// old log segments to free up space. A value of 0 indicates no limit.
+	RetentionMaxMessages int64
+	// The TTL for stream log segment files, after which they are deleted. A value of 0 indicates no TTL.
+	RetentionMaxAge int64
```
Comment (tylertreat) [inline client.go:106]: We should use `time.Duration` and handle converting it to the int64. Using `time.Duration` provides a nicer API experience to the end user IMO.

File: client.go [hunk 3]
```diff
@@ -88,6 +95,30 @@ type StreamOptions struct {
...
+	// The frequency to check if a new stream log segment file should be rolled and whether any segments are
+	// eligible for deletion based on the retention policy or compaction if enabled.
+	CleanerInterval int64
```
Comment (tylertreat) [inline client.go:109]: Same thing here.

File: client.go [hunk 4]
```diff
@@ -88,6 +95,30 @@ type StreamOptions struct {
...
+	// The maximum size of a single stream log segment file in bytes. Retention is always done a file at a time,
+	// so a larger segment size means fewer files but less granular control over retention.
+	SegmentMaxBytes int64
+	// The maximum time before a new stream log segment is rolled out. A value of 0 means new segments will only
+	// be rolled when segment.max.bytes is reached. Retention is always done a file at a time,
+	// so a larger value means fewer files but less granular control over retention.
+	SegmentMaxAge int64
```
Comment (tylertreat) [inline client.go:116]: Same here.

File: message.go [hunk 5]
```diff
@@ -41,6 +42,8 @@ func (a AckPolicy) toProto() proto.AckPolicy {
 	return proto.AckPolicy(a)
 }
 
+// CompactEnabled controls stream compaction configuration
```
Comment (tylertreat) [inline message.go:45]: Should be removed?

File: client.go (hunk 2 above)
Comment (LaPetiteSouris) [reply client.go:99]: Agree. It's more natural to configure in `time.Duration`. the API should handle the conversion behind the scene

**Pull request merged**